File: VB/BeginEndDataUpdate/DataItem.vb
Imports DevExpress.Mvvm
Imports System.ComponentModel

Namespace BeginEndDataUpdate

    Public Class DataItem
        Inherits BindableBase

        Public Property Name As String
            Get
                Return GetValue(Of String)(NameOf(DataItem.Name))
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property Value As Integer
            Get
                Return GetValue(Of Integer)(NameOf(DataItem.Value))
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return Name
        End Function
    End Class
End Namespace

File: CS/BeginEndDataUpdate/DataItem.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BeginEndDataUpdate {
    public class DataItem : BindableBase {
        public string Name {
            get { return GetValue<string>(nameof(Name)); }
            set { SetValue(value); }
        }

        public int Value {
            get { return GetValue<int>(nameof(Value)); }
            set { SetValue(value); }
        }

        public override string ToString() {
            return Name;
        }
   }
}
